opds.py

```diff
@@ -148,16 +148,23 @@ def __init__(self, _db, title, url, libraries, annotator=None):
         self.annotator = annotator
        
         for library in libraries:
-            self.feed.append(self.library_entry(library))
+            if not isinstance(library, tuple):
+                library = (library,)
+            self.feed.append(self.library_entry(*library))
         annotator.annotate_feed(self)
-            
+        
     @classmethod
-    def library_entry(cls, library):
+    def library_entry(cls, library, distance=None):
         entry = AtomFeed.entry(
             AtomFeed.id(library.urn_uri),
             AtomFeed.title(library.name),
             AtomFeed.updated(cls._strftime(library.timestamp))
         )
+
+        if distance is not None:
+            distance_tag = AtomFeed.SCHEMA.distance()
+            distance_tag.text = "%d km." % (distance/1000)
+            entry.append(distance_tag)
```

Comment:
Yeah, it's a possibility. It doesn't really cost anything to include this information, and making it visible would explain why the results are in the order they are.